Subject: [PATCH] powerpc: powermac adb fix dependency on btext_drawchar
From: Andy Whitcroft <apw@shadowen.org>
Date: 1133372090 +0000

udbg_adb_init() has become dependent on btext_drawchar, even when
BOOTX_TEXT support is not selected.  This leads to the error below.
Make the check dependant on BOOTX_TEXT.

      LD      .tmp_vmlinux1
    arch/powerpc/platforms/built-in.o(.toc1+0xa40): undefined
					reference to `btext_drawchar'

Signed-off-by: Andy Whitcroft <apw@shadowen.org>
Signed-off-by: Paul Mackerras <paulus@samba.org>
Signed-off-by: Olaf Hering <olh@suse.de>

---

 arch/powerpc/platforms/powermac/udbg_adb.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

f8809d0b4ebc657afc0924910eca632dee5cfe46
diff --git a/arch/powerpc/platforms/powermac/udbg_adb.c b/arch/powerpc/platforms/powermac/udbg_adb.c
index e51de55..d1c40e2 100644
--- a/arch/powerpc/platforms/powermac/udbg_adb.c
+++ b/arch/powerpc/platforms/powermac/udbg_adb.c
@@ -171,9 +171,12 @@ int udbg_adb_init(int force_btext)
 	udbg_adb_old_getc_poll = udbg_getc_poll;
 
 	/* Check if our early init was already called */
-	if (udbg_adb_old_putc == udbg_adb_putc ||
-	    udbg_adb_old_putc == btext_drawchar)
+	if (udbg_adb_old_putc == udbg_adb_putc)
 		udbg_adb_old_putc = NULL;
+#ifdef CONFIG_BOOTX_TEXT
+	if (udbg_adb_old_putc == btext_drawchar)
+		udbg_adb_old_putc = NULL;
+#endif
 
 	/* Set ours as output */
 	udbg_putc = udbg_adb_putc;
-- 
1.0.6
